py/configu/commands/eval_command.py

```diff
@@ -158,24 +158,36 @@ def _validate_result(self, result: EvalCommandReturn):
             for key, value in result.items():
                 cfgu = value["context"]["cfgu"]
                 evaluated_value = value["result"]["value"]
-                type_test = ConfigSchema.CFGU.VALIDATORS.get(
-                    cfgu.type.value, lambda: False
-                )
-                test_values = (
-                    (
-                        evaluated_value,
-                        cfgu.pattern,
+                try:
+                    type_test = ConfigSchema.CFGU.VALIDATORS[cfgu.type.value]
+                    test_values = (
+                        (
+                            evaluated_value,
+                            cfgu.pattern,
+                        )
+                        if cfgu.type == CfguType.REG_EX
+                        else (evaluated_value,)
                     )
-                    if cfgu.type == CfguType.REG_EX
-                    else (evaluated_value,)
-                )
-                if not type_test(*test_values):
-                    raise ValueError(
+                    if not type_test(*test_values):
+                        raise ValueError(
+                            error_message(
+                                f"invalid value type for key '{key}'", error_scope
+                            ),
+                            f"value '{test_values[0]}' must be a " f"'{cfgu.type}'",
+                        )
+                except KeyError:
+                    raise KeyError(
                         error_message(
-                            f"invalid value type for key '{key}'", error_scope
+                            "invalid type property", error_scope + [key, "type"]
                         ),
-                        f"value '{test_values[0]}' must be a " f"'{cfgu.type}'",
+                        f"type '{cfgu.type.value}' is not yet supported in this SDK. "
+                        f"For the time being, please utilize the String type. "
+                        f"We'd greatly appreciate it if you could open an issue "
+                        f"regarding this at "
+                        f"https://github.com/configu/configu/issues/new/choose "
+                        f"so we can address it in future updates.",
                     )
+
```

Comment:
In cases such as these I tend to recommend the following pattern:
```py
try:
    <<potentially raising code>>    
except KeyError as e:
    raise <<new exception>> from e
```
That way the underlying stacktrace is persisted and the user gets more detailed error messaging, making it easier to debug.

Additionally, do all lines in the error message have to be f-strings? If not we should probably only make the first line an f-string (not sure if that's supported by python syntax :thinking:).

Reply:
Syntax supported.
Fixed.
Thanks